orm/query.go

```diff
@@ -812,6 +812,46 @@ func (q *Query) DropTable(opt *DropTableOptions) (Result, error) {
 	})
 }
 
+func (q *Query) HasTable() (bool, error) {
+	if q.stickyErr != nil {
+		return false, q.stickyErr
+	}
+
+	var count int
+	_, err := q.db.QueryOne(Scan(&count), hasTableQuery{
+		q: q,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if count > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (q *Query) HasColumn(clm string) (bool, error) {
+	if q.stickyErr != nil {
+		return false, q.stickyErr
+	}
+
+	var count int
+	_, err := q.db.QueryOne(Scan(&count), hasColumnQuery{
+		q: q,
+		clmName: clm,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	if count > 0 {
+		return true, nil
+	}
+
+	return false, nil
+}
```

Comment:
Let's not add these 2 methods to the `*Query` since it clutters API and is not very useful, e.g. `db.Model(&strct).Where().HasTable` does not make much sense. You should be able to move implementation from `Query.HasTable` to `orm.HasTable` without any issues...

Reply:
I think CreateTable and DropTable should be move too.